sklearn/tree/tests/test_tree.py

```diff
@@ -1266,3 +1268,39 @@ def check_min_weight_leaf_split_level(name):
 def test_min_weight_leaf_split_level():
     for name in ALL_TREES:
         yield check_min_weight_leaf_split_level, name
+
+
+def check_public_apply(name):
+    X_small32 = X_small.astype(tree._tree.DTYPE)
```

Comment:
The goal is to test whether the public apply returns the same result as the private apply `est.tree_.apply`, which doesnt do any checking.